2-3/app.js

```diff
@@ -1,59 +1,111 @@
 const btn = document.getElementById('btn');
-const target = document.getElementById('task')
-const todolist = []
+const target = document.getElementById('task');
+const todolist = [];
+const workstatus = document.getElementById('workstatus');
+const donestatus = document.getElementById('donestatus');
+const allstatus = document.getElementById('allstatus');
+
+workstatus.addEventListener('input', () => {
+    const worktask = todolist.filter(todo => {
+        return (todo.status == '作業中')
+    })
+    showtodo(worktask)
+})
+
+donestatus.addEventListener('input', () => {
+    const donetask = todolist.filter(todo => {
+        return (todo.status == '完了')
+    })
+    showtodo(donetask)
+})
+
+allstatus.addEventListener('input', () => {
+    showtodo()
+})
 
 btn.addEventListener('click', () => {
     const newtask = document.getElementById('new-task');
-    const task = { name: newtask.value, status: '作業中' };
+    const task = { id: "", name: newtask.value, status: '作業中' };
     todolist.push(task);
-    newtask.textContent = '';
-    showtodo();
+    newtask.value = '';
+    if (donestatus.checked) {
+        todolist.filter(todo => {
+            todolist[todolist.length - 1].id = todolist.length
+        })
+    } else if (workstatus.checked) {
+        const worktask = todolist.filter(todo => {
+            todolist[todolist.length - 1].id = todolist.length
+            return (todo.status == '作業中')
+        })
+        showtodo(worktask)
+    } else { showtodo(); }
```

Comment:
以下のようにすれば `showtodo` の `else` 以下の処理が無くせそうな気がします。

```suggestion
    } else { showtodo(todolist); }
```